src/librustc/ty/subst.rs

```diff
@@ -298,20 +275,20 @@ impl<'a, 'gcx, 'tcx> Substs<'tcx> {
                        target_substs: &Substs<'tcx>)
                        -> &'tcx Substs<'tcx> {
         let defs = tcx.lookup_generics(source_ancestor);
-        Substs::new(tcx, target_substs.iter().chain(&self[defs.own_count()..]).cloned())
+        tcx.mk_substs(target_substs.iter().chain(&self[defs.own_count()..]).cloned())
```

Comment:
It's not but that's irrelevant here. `cloned` is not bad, it's just a sign of a slice iterator, which in some cases (not this) can be replaced with the slice itself.